test/end-to-end/helpers/JSWait.js

```diff
@@ -1,23 +1,30 @@
+const {decodeHTML} = require('test/end-to-end/helpers/GeneralHelpers');
+
 class JSWait extends codecept_helper {
 
-    _beforeStep(step) {
+    async _beforeStep(step) {
 
         const helper = this.helpers.WebDriver || this.helpers.Puppeteer;
 
         if (step.name === 'seeCurrentUrlEquals' || step.name === 'seeInCurrentUrl') {
             return helper.waitForElement('body', 30);
         }
+
+        if (step.name === 'waitForText') {
+            // this handles decoding any HTML coded characters in the text
+            step.args[0] = await decodeHTML(step.args[0].trim());
+        }
     }
 
     async navByClick(text, locator = null, webDriverWait = 2) {
         const helper = this.helpers.WebDriver || this.helpers.Puppeteer;
         const helperIsPuppeteer = this.helpers.Puppeteer;
 
         if (helperIsPuppeteer) {
-            await Promise.all([
-                helper.page.waitForNavigation({waitUntil: ['domcontentloaded', 'networkidle0']}),
-                locator ? helper.click(text, locator) : helper.click(text)
-            ]);
+            helper.click(text, locator).catch(err => {
+                console.error(err.message);
```

Comment:
I feel like I tried adding an await here in previous PRs for probate-caveats-frontend and it caused issues. I've tested this ALOT over the last 48hrs and it's been running consistently, so it might be worth looking into this in the same PR as Phani's change, if that's ok? And get the benefit of all the other fixes in this PR?

Reply:
ok